### PR TITLE
Migrate `transform:` callsites and introduce breaking version of GDT (9.0)

### LINKED ISSUE #7177: App store binary warning received due to use of private API transform: by GoogleTransport
### Step 0: Are you in the right place?

  * For issues or feature requests related to __the code in this repository__
    file a Github issue.
    * If this is a __feature request__ please use the Feature Request template.
  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the `firebase` tag.
  * For general (non-iOS) Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For backend issues, console issues, and other non-SDK help that does not fall under one
    of the above categories, reach out to
    [Firebase Support](https://firebase.google.com/support/).
  * Once you've read this section and determined that your issue is appropriate for
    this repository, please delete this section.

### [REQUIRED] Step 1: Describe your environment

  * Xcode version: ___12.2__
  * Firebase SDK version: __20201104 download, said to have GoogleTransport 8.0.0___
  * Installation method: Zip file from https://firebase.google.com/docs/ios/setup#frameworks
  * Firebase Component: __GoogleTransport (required for Analytics)___ 

#### Steps to reproduce:

1. Incorporate frameworks needed for Analytics to project as described in README.md in the zip.
2. Invoke the functionality by `#import "Firebase.h"` and  `[FIRApp configure]` in AppDelegate.m
3. Build the project for Generic iOS Device and submit to App Store from Xcode
4. Binary is accepted and can be used for TestFlight, but warning "app references non-public selectors in Payload: transform:" is popped up and stored in the upload load.
5. I have every expectation that this will result in rejection of the binary at app review stage, at which point I'll have no choice but to remove the Analytics functionality completely from my app.

#### Relevant Code:

The 'transform:' use is in the GoogleTransport framework which is said by the README to be required for Analytics functionality.

There is an earlier closed and uncommentable issue here:

https://github.com/firebase/firebase-ios-sdk/issues/5568

There it says this use of transform: is not just a chance reuse of the same name, but an actual call to Apple's private API, which is indeed wrong. The issue was closed saying Apple seemed to stop checking for this, however I can confirm that as of December 8, 2020, they had apparently started again. In any case this use is a violation of Apple's rules and should not be present in a public SDK intended for app developer use.

### PR DESCRIPTION
**Pull request by ncooke3**
• There is an [upcoming breaking change in GDT (8.4 -> 9.0)](https://github.com/google/GoogleDataTransport/commit/0afaad7819fe7cd66d772ddebb093b89eea3b2a9) where the `transform:` API has been renamed to `transformGDTEvent:` to [avoid collisions with one of Apple's private APIs](#7177).

This PR prepares for this change by:
• Migrating callsites to the new API
• Updating specs to point to next major version of GDT (9.0)

This is planned to be a part of Firebase 8.

Fixes #7177

### CODE REVIEW

**Comment by ncooke3** [comment]
For this type of change, which CHANGELOGs should be modified? The FirebaseCore/CHANGELOG or the CHANGELOGs that correspond to specs that have been changed?

**Review by paulb777 — CHANGES_REQUESTED** [review]
Please push GDT 9.0 to SpecsDev and rerun tests

Only one overall Firebase changelog is needed in the FirebaseCore CHANGELOG

**Comment by paulb777** [comment]
To fix the CI failure, https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseMLModelDownloader/Apps/Sample/Podfile should be updated to get FirebaseCoreDiagnostics from the local repo - similar syntax to https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseMLModelDownloader/Apps/Sample/Podfile#L8

**Review by paulb777 — APPROVED** [review]

File: FirebaseMLModelDownloader/Apps/Sample/Podfile [hunk 1]
```diff
@@ -6,6 +6,7 @@ source 'https://github.com/firebase/SpecsStaging.git'
 source 'https://cdn.cocoapods.org/'
 
 pod 'FirebaseMLModelDownloader', :path => '../../../'
+pod 'FirebaseCoreDiagnostics', :path => '../../../'
```
Comment (ncooke3) [inline FirebaseMLModelDownloader/Apps/Sample/Podfile:9]: Fixed CI failure

**Comment by maksymmalyhin** [comment]
cc @visumickey

**Review by maksymmalyhin — APPROVED** [review]
LGTM

**Review by visumickey — APPROVED** [review]
Thanks for making this change. LGTM!

**Comment by ncooke3** [comment]
Can address any additional feedback in a subsequent PR.

**Pull request merged**